cupy/_core/core.pyx

```diff
@@ -2406,11 +2406,15 @@ cpdef _ndarray_base array(obj, dtype=None, bint copy=True, order='K',
     concat_shape, concat_type, concat_dtype = (
         _array_info_from_nested_sequence(obj))
     if concat_shape is not None:
-        # TODO(leofang): handle blocking?
         if blocking:
-            raise NotImplementedError
+            warnings.warn(
+                'blocking is set to True, but it is unnecessary in the '
+                'current version for handling nested sequence inputs; a '
+                'redundant synchronization would happen',
+                _util.PerformanceWarning)
```

Comment:
Do we need this warning? The `blocking` argument might make sense in some cases, e.g.:

```py
data = [cupy_arr1, cupy_arr2]
x = cupy.array(data)  # users might want to pass blocking=True here
...
with stream1:
    cupy_arr1.fill(0)
```

Reply:
In this case it would be better that a stream order is established instead of adding synchronization, i.e.
```py
data = [cupy_arr1, cupy_arr2]
x = cupy.array(data)  # blocking = False
...
e = cp.cuda.get_current_stream.record()
stream1.wait(e)
with stream1:
    cupy_arr1.fill(0)
```
The problem is it's not easy to find a balance in making the warning message detailed yet succinct 😅

Reply:
Also `blocking=True` is unnecessary if we're concatenating NumPy arrays (i.e. `data = [numpy_arr1, numpy_arr2]`), this was my original thinking.

Reply:
Perhaps to not warn at all (as you suggested) would save us the trouble from coming up a good warning message...? 😂

Reply:
Removed in commit 6b962f7283e4870811e0024163f42063cfe32f30